cogito/protocol.go

```diff
@@ -137,6 +151,19 @@ func (ver Version) String() string {
 	return fmt.Sprint("ref: ", ver.Ref)
 }
 
+// Output is the JSON object emitted by the get and put step.
+type Output struct {
+	Version Version `json:"version"`
+	// the Cogito resource doesn't use the optional "metadata" field.
+	// Metadata Metadata `json:"metadata"`
+}
+
+// GetParams is the "params:" block in a pipeline get step for the Cogito resource.
+// Cogito doesn't accept any get step parameters, so to give a slightly better error
+// message form the JSON parsing, instead of declaring it an empty struct, we do not
```

Comment:
`form` or `for`?

Reply:
@aliculPix4D 

> `form` or `for`?

3rd possibility: `from` :-). Addressed in last PR in this series, thanks.

>  `so to give a slightly better error message from the JSON parsing`.
> Why and how this gives a better msg?

Because in Go (and in any statically typed language), we can leave the compiler or the runtime do the type validation for us. In this case it is at runtime (since the input comes at runtime).

Imagine that the pipeline had a get step for cogito (forget for a moment that get does nothing for cogito):

```yaml
...
get: gh-status
params: {foo: bar}
```

since the get request struct is:

```go
type GetRequest struct {
	Source  Source  `json:"source"`
	Version Version `json:"version"`
	// Cogito does not support get params; a resource supporting them would have the
	// following line uncommented:
	// Params  GetParams `json:"params"`
	Env Environment
}
```

so, no `GetParams` field, it would the JSON parser itself to return an error, no need for our code (as we were doing with `ofcourse`) to do any input validation at _this_ level.